app/Actions/User/Notify.php

```diff
@@ -22,6 +22,10 @@ public function do(Photo $photo): void
 		$users = new Collection([User::query()->find(0)]);
 		$album = $photo->album;
 		if ($album !== null) {
+			// phpstan throws this error
+			// Parameter #1 ...$values of method Illuminate\Support\Collection<int,Illuminate\Database\Eloquent\Model|null>::push() expects Illuminate\Database\Eloquent\Model|null, Illuminate\Database\Eloquent\Collection given.
+			// However the signature is push(... $values) which also consider collection/array...
+			// @phpstan-ignore-next-line
 			$users->push($album->shared_with);
 			$users->push($album->owner);
```

Comment:
PHPstan is right here. Allthough the signature is `push(...$values)`, this does not imply that you can pass an array directly. A variable-length parameter means that you can do something like `push($model1, $model2, $model3)` and then these tree models are available as as an array `$values` inside the function. However, if you call `push($models)` and `$models` is an array, then `$values` is an array with a single element which contains the array `$models`. That is not what we want here.

There are several solutions to this problem discussed in https://stackoverflow.com/questions/9045366/php-pass-array-as-variable-length-argument-list and https://stackoverflow.com/questions/744145/passing-an-array-as-arguments-not-an-array-in-php. I don't like neither of them. They are all ugly as hell.

Another solution would be to use [`Collection::concat`](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Collections/Collection.php#L935). Unfortunately, `concat` returns a new `Collection` and does not concatenate the collections in place, so we would need something like `$users = $users->concat($album->shared_width)`. But this is probably still the best solution.

Reply:
Fixed [4b1ff0b](https://github.com/LycheeOrg/Lychee/pull/1469/commits/4b1ff0be6a835e98c9f532d00fcf75bc55ff2efb)